### ISSUE TITLE
displaying totalRuns instead of totalFeatures would provide a better overview in index.html

### ISSUE THREAD

**Issue by igalsfy** [renatoathaydes/spock-reports#182]
Hi,

today, index.html displays the number features run in a class which generates 2 types of inconsistencies:
- if class A extends class B and def methods are implemented in classe B, class A will display 0 features run (one could read it as "0 tests run"
- if a def method uses the Unroll keyword, only one feature will be listed and more tests were in fact ran.

So, I propose to provide in the index.html report the totalRuns instead of the totalFeatures. I've done the patch locally and it provides more meaningful results IMO.

What do you think?

**Comment by renatoathaydes** [comment]
Hi @igalsfy , thanks for your interest in helping.

Regarding the first point: I don't think that's correct. If class `A` has no features but it extends `Specification`, then it should show in the report and this is expected, even if it has 0 features. Because that's true.

If you don't want it in the report, make it `abstract`.

More commonly, you would have some features in both classes... I have some tests that extend other tests and I see the expected feature numbers in the summary report for them. If you are convinced there's something wrong, please provide a sample with the behaviour you believe is incorrect in a new issue, so I can have a look at it.

About the second point: this seems to be a common misconception. The fact that you have many examples in a test doesn't make it become several tests. Nearly every test you write should contain at least a few examples... but the examples are, or should be, used to test a single function of your code, so they are still part of a single test. 

I need to ask you: what do you mean by `def method`? `def` is just a Groovy keyword for defining a variable or method, it doesn't make either special in any way... so I'm confused what you are trying to say.

**Comment by igalsfy** [comment]
Hi,
Thank you for your feedback.

Regarding both points,

I am using your plugin to make high level reports of test execution.

My framework works in such a way:

- class `TestSpec `extends `Specification`.
- `TestSpec `is a class containing one test with this kind of implementation:

```
@Unroll
def "#config"() {
    when: 
        println("hello" + #config)
    then:
        assert true
    where:
        config << thisIsAList()
}
```

- class `RealTest `extends `TestSpec`
- `RealTest `is a class containing a constructor for the list passed to the method` "#config"()` which is `thisIsAList()`
- I have several classes similar to `RealTest `which provide different kinds of lists for `"#config"` to run on.
- I only run the class `RealTest `in my framework.

My problem is that when I display my report `index.html`, it show that 0 features have been run `RealTest`, when in fact, hundreds of test cases could have been run in the process and there is no feedback about it in `index.html`.

Furthermore, the junit xml provided by the spock framework correctly list the number of test cases run, so my CI offers 2 different results :

JUnit says that, for instance, 100 tests have been run on `RealTest`, and the index.html says that 0 features have been run on `RealTest`. 

I know I can then click on the `RealTest `class hyperlink and end up providing the correct feedback from execution, but at a high level, this doesn't work for me.

So basically, I would rather display `totalRuns `instead of `totalFeatures `in index.html but I have found no way to override it using the numerous tools you provide to generate custom reports. The only solution I've found is to clone the project and make the modification in the groovy myself.

I hope I've made myself clearer :)

Thank you.

**Comment by Auliyaa** [comment]
Here is a sample without @Unroll that just defines a test in an abstract top-class:

`
class TestSpec extends AbstractSpec {
    def setup() {
        println("setting-up")
    }
}
`
`
import spock.lang.Specification

class AbstractSpec extends Specification {
    def "My test case"() {
        when:
        def a = 1+1
        then:
        a == 2
    }
}
`

And the HTML report looks like this:
![image](https://user-images.githubusercontent.com/6253590/70793518-c1fde600-1d9b-11ea-9694-12982ecaeddb.png)

**TestSpec** has 0 feature and AbstractSpec does not appear inside the report itself.
It seems that in such a case, all features from the abstract class should appear as features from the implementation. Or that we should at least see the abstract class appear inside the report itself.

**Comment by renatoathaydes** [comment]
Your latest example seems like a real issue, but separate from this one. Do you mind copying the above into a new issue so we can keep track there?

**Comment by Auliyaa** [comment]
Sorry for the delay, I've move the second example inside a separate issue here: #187
